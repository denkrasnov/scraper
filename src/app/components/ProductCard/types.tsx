import { Product } from "src/app/services/fetchProducts/types";

export interface ProductCardProps
  extends Pick<Product, "imageUrl" | "title" | "price"> {}
