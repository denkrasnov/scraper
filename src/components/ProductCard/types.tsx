import { Product } from "src/services/fetchProducts/types";

export interface ProductCardProps
  extends Pick<Product, "imageUrl" | "title" | "price" | "noImage"> {}
