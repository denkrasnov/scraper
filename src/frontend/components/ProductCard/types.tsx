export interface ProductCardProps {
  header?: string;
  imageURL?: string;
  newsURL?: string;
  date?: string;
}
