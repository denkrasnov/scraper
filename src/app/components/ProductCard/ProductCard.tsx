import React, { FC } from "react";

import Image from "src/atoms/Image";
import Card from "src/atoms/Card";
import Box from "src/atoms/Box";
import { H2 } from "src/atoms/Headings";
import Text from "src/atoms/Text";
import Divider from "src/atoms/Divider";
import { ProductCardProps } from "./types";

const ProductCard: FC<ProductCardProps> = props => {
  const { title, imageUrl } = props;

  return (
    <Card>
      <Box flexDirection="column" height="100%">
        <Box height="200px">
          <Image alt="product photo" src={imageUrl} />
        </Box>
        <Box flexDirection="column" height="100%" padding="0 10px 10px 10px">
          <H2>{title}</H2>
          <Box flex={1} />
          <Divider />
          <Box margin="15px 0px 5px 0">
            <Text color="MAIN_RED" fontSize="fs400">
              Price here
            </Text>
          </Box>
        </Box>
      </Box>
    </Card>
  );
};

export default ProductCard;
