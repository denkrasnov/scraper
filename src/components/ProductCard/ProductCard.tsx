import React, { FC } from "react";

import Image from "src/atoms/Image";
import Card from "src/atoms/Card";
import Box from "src/atoms/Box";
import { H3 } from "src/atoms/Headings";
import Text from "src/atoms/Text";
import Divider from "src/atoms/Divider";
import { ProductCardProps } from "./types";
import placeholder from "./images/placeholder.jpg";

const ProductCard: FC<ProductCardProps> = props => {
  const { title, imageUrl, price, noImage } = props;

  return (
    <Card>
      <Box flexDirection="column" height="100%">
        <Box height="200px">
          <Image alt="product photo" src={noImage ? placeholder : imageUrl} />
        </Box>
        <Box flexDirection="column" height="100%" padding="0 10px 10px 10px">
          <H3>{title}</H3>
          <Box flex={1} />
          <Divider />
          <Box justifyContent="flex-end" margin="15px 0px 5px 0">
            <Text color="MAIN_RED" fontSize="fs400" fontWeight="fw700">
              lei {price}
            </Text>
          </Box>
        </Box>
      </Box>
    </Card>
  );
};

export default ProductCard;
