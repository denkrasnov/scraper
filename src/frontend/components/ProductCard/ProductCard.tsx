import React, { FC } from "react";

import Image from "~app/atoms/Image";
import Card from "~app/atoms/Card";
import Box from "~app/atoms/Box";
import { H3 } from "~app/atoms/Headings";
import Text from "~app/atoms/Text";
import Divider from "~app/atoms/Divider";
import { ProductCardProps } from "./types";
import { image } from "~app/assets";

const ProductCard: FC<ProductCardProps> = (props) => {
  const { title, imageUrl, price, noImage } = props;

  return (
    <Card>
      <Box flexDirection="column" height="100%">
        <Box height="200px">
          <Image
            alt="product photo"
            src={noImage ? image.tvPlaceholder : imageUrl}
          />
        </Box>
        <Box flexDirection="column" height="100%" padding="0 10px 10px 10px">
          <H3>{title}</H3>
          <Box flex={1} />
          <Divider />
          <Box justifyContent="flex-end" margin="15px 0px 5px 0">
            <Text color="MAIN_RED" fontSize="fs400" fontWeight="fw700">
              lei {price}
            </Text>
          </Box>
        </Box>
      </Box>
    </Card>
  );
};

export default ProductCard;
