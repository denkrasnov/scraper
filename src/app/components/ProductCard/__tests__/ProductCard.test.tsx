import React from "react";
import { shallow } from "enzyme";

import ProductCard from "..";

describe("ProductCard", () => {
  it("should render", () => {
    const component = shallow(
      <ProductCard imageUrl="__IMAGE_URL__" title="__TITLE__" />
    );
    expect(component).toMatchSnapshot();
  });
});
