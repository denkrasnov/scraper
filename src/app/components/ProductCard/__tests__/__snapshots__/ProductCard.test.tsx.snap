// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCard should render 1`] = `
<Box
  backgroundColor="WHITE"
  display="block"
  maxWidth="380px"
>
  <Card>
    <Box
      flexDirection="column"
    >
      <Box
        height="200px"
      >
        <Image
          alt="product photo"
          src="__IMAGE_URL__"
        />
      </Box>
      <Box
        flexDirection="column"
        padding="0 10px 10px 10px"
      >
        <Component>
          __TITLE__
        </Component>
        <Box
          margin="0 0 15px"
        >
          <Text>
            Description here
          </Text>
        </Box>
        <Divider />
        <Box
          margin="15px 0px 5px 0"
        >
          <Text
            color="MAIN_RED"
            fontSize="fs400"
          >
            Price here
          </Text>
        </Box>
      </Box>
    </Box>
  </Card>
</Box>
`;
